flapison/schema.py

```diff
@@ -47,25 +47,47 @@ def compute_schema(schema_cls, default_kwargs, qs, include):
             if "." in include_path:
                 related_includes[field] += [".".join(include_path.split(".")[1:])]
 
-    # make sure id field is in only parameter unless marshamllow will raise an Exception
-    if schema_kwargs.get("only") is not None and "id" not in schema_kwargs["only"]:
-        schema_kwargs["only"] += ("id",)
+    # manage only parameter of the schema
+    only = schema_kwargs.get("only")
 
-    # create base schema instance
-    schema = schema_cls(**schema_kwargs)
+    # collect sub-related_only
```

Comment:
I don't like overloading `schema_kwargs` here. It's supposed to be a dictionary that's passed directly into the schema constructor, with all the same fields and meanings as defined by `marshmallow-jsonapi`. By doing this we're adding extra magic for related fields. I agree modifying `only` for related resources should be possible, but maybe but it could be done using a new property on the resource, like `related_kwargs` or `get_related_kwargs`,, which is a function that takes the related resource name and returns the kwargs for this resource.

Reply:
I agree about that use a new property on the resource like `related_kwargs` declare the related schema init attributes directly.
But because of we can define `only` or `exclude` in the schema like
```python
class PostSchema(Schema):
    tags = Relationship(TagSchema(only=('name',)), type_='tags, many=True)
``` 
should we consider the `only` in this situation? I feel confuse

Reply:
I don't like it, but I think you can override the `only` field of the `Relationship` on the `Schema` *instance* after it's been constructed. And this won't affect any other schema instances because all the declared fields are copied when the a schema instance is created. 

So if the users had a hook that passed in the schema instance, they could say:
```python
schema.declared_fields.tags.only = ('some', 'dynamic', 'values')
```

Currently no hooks let you access the schema, but perhaps we could create a `before_dump` hook that passes in the schema instance, in order to be modified?

Reply:
Good point.